### PR TITLE
Enhancement: Deterministic Module Configuration

### LINKED ISSUE #64: Deterministic Configuration
At present EOM configures itself when the module is loaded. However, if the ```oslo.config.cfg.CONF``` is initialized with the project name after the module is loaded then it may not receive its settings. This is resolvable by adding a simple ```configure()``` to each of the EOM modules that is called after the module is loaded in order to get the data from the configuration file at a deterministic point.

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
- Add a ```configure()``` to each EOM Middleware Module
- Update documentation to show how to configure the EOM Middleware
- Update tests to properly configure the EOM Middleware
- Updated EOM Middleware for consistency in layout
- Updated the logging.conf-sample to have a valid configuration
- Updated the eom.conf-sample for the logging configuration

Closes #64

### CODE REVIEW

File: etc/logging.conf-sample [hunk 1]
```diff
@@ -1,40 +1,48 @@
...
+[handler_console]
+level = DEBUG
+class = logging.StreamHandler
+formatter = standard
+
+[handler_logfile]
+level = DEBUG
+class = logging.FileHandler
+formatter = standard
+filename = /home/bmeyer/.eom/log/eom.log
```
Comment (TheSriram) [inline etc/logging.conf-sample]: can we change these to be more generic?

File: etc/logging.conf-sample [hunk 2]
```diff
@@ -1,40 +1,48 @@
...
+level = DEBUG
+class = logging.handlers.SysLogHandler
+formatter = standard
+address = /dev/log
+
+[handler_rotatelogfile]
+level = DEBUG
+class = logging.handlers.RotatingFileHandler
+formatter = standard
+args=('/home/bmeyer/.eom/log/eom.log', 400000000, 2)
```
Comment (TheSriram) [inline etc/logging.conf-sample]: can we change these to be more generic

File: etc/logging.conf-sample (hunk 1 above)
Comment (BenjamenMeyer) [inline etc/logging.conf-sample]: Thought I caught those...will update

File: etc/logging.conf-sample (hunk 2 above)
Comment (BenjamenMeyer) [inline etc/logging.conf-sample]: same https://github.com/racker/eom/pull/65#discussion_r32129488

**Comment by TheSriram** [comment]
:+1:

**Pull request merged**